### ISSUE TITLE
Add (display) window icon

### ISSUE THREAD

**Issue by danfe** [willbprog127/spiritvnc-fltk#2]
SpiritVNC comes with nice colorful `spiritvnc.xpm` but it's currently unused.  I've ported the program to FreeBSD and added couple of lines to make use of the icon, perhaps you might want to merge some of those [changes](https://cgit.freebsd.org/ports/commit/?id=7034613152e365155bf3cbe0aa4bef0d1ba9514c).

**Comment by willbprog127** [comment]
@danfe Thanks, will check it out! 👍

**Comment by willbprog127** [comment]
I don't have a FreeBSD VM available yet, so can't test changes all around, but Linux definitely doesn't like it:

```c
    #ifndef __APPLE__
    // set window's icon
    #include <X11/xpm.h>
    fl_open_display();                // needed if display has not been previously opened

    Pixmap pm;
    Pixmap mask;
    XpmCreatePixmapFromData(fl_display, DefaultRootWindow(fl_display), pmSpiritvnc_xpm, &pm, &mask, NULL);

    app->mainWin->icon(reinterpret_cast<void *>(pm));
    #endif
```
...yields the following error during compile:

```
In file included from /usr/include/X11/Xlib.h:47,
                 from /usr/include/FL/x.H:37,
                 from /usr/include/FL/fl_draw.H:27,
                 from src/app.h:43,
                 from src/spiritvnc.cxx:47:
/usr/include/X11/xpm.h: In function ‘int main(int, char**)’:
/usr/include/X11/xpm.h:286:1: error: expected unqualified-id before string constant
 _XFUNCPROTOBEGIN
 ^~~~~~~~~~~~~~~~
src/spiritvnc.cxx:95:5: error: ‘XpmCreatePixmapFromData’ was not declared in this scope
     XpmCreatePixmapFromData(fl_display, DefaultRootWindow(fl_display), pmSpiritvnc_xpm, &pm, &mask, NULL);
     ^~~~~~~~~~~~~~~~~~~~~~~
src/spiritvnc.cxx:95:5: note: suggested alternative: ‘XCreatePixmapFromData’
     XpmCreatePixmapFromData(fl_display, DefaultRootWindow(fl_display), pmSpiritvnc_xpm, &pm, &mask, NULL);
     ^~~~~~~~~~~~~~~~~~~~~~~
     XCreatePixmapFromData
```
I tried `XCreatePixmapFromData` but it just gives me a similar error.  I'll keep working on it when I have time...

**Comment by danfe** [comment]
Hmm, just tried it here on Ubuntu 20.04.1, builds fine.  `_XFUNCPROTOBEGIN` is defined in the `X11/Xfuncproto.h` which is included by the `X11/Xlib.h` which is, in turn, included by the `X11/xpm.h`.  Are all of these files present on your system and not corrupted?  Which GNU/Linux distribution are you using?

**Comment by willbprog127** [comment]
> Hmm, just tried it here on Ubuntu 20.04.1, builds fine. `_XFUNCPROTOBEGIN` is defined in the `X11/Xfuncproto.h` which is included by the `X11/Xlib.h` which is, in turn, included by the `X11/xpm.h`. Are all of these files present on your system and not corrupted? Which GNU/Linux distribution are you using?

I'm using antiX Linux.  I'll probably need to create Ubuntu and FreeBSD virtual machines to test.  Yes, those files are on my system and not corrupted.  I grepped for the definitions and they were there.
Thanks! :+1:

**Comment by willbprog127** [comment]
Post on FLTK's Google Groups about the compile issues: https://groups.google.com/g/fltkgeneral/c/hcjV-rgjHWM/m/5CCLU-rMAgAJ

**Comment by willbprog127** [comment]
Fixed with commit ac5d963154333cd5b35813a046e6a3f3116da7d6
The problem compiling was due to the `#include <X11/xpm.h>` directive not being above the other `#include <FL...` directives.  Also included Ian MacArthur's transparency code.
